src/wp-includes/class-wp-textdomain-registry.php

```diff
@@ -84,7 +84,25 @@ public function get( $domain, $locale ) {
 	 * @return bool Whether any MO file paths are available for the domain.
 	 */
 	public function has( $domain ) {
-		return ! empty( $this->current[ $domain ] ) || empty( $this->all[ $domain ] );
+		if ( ! empty( $this->current[ $domain ] ) || empty( $this->all[ $domain ] ) ) {
+			return true;
+		}
+
+		$locations = $this->get_paths_for_domain( $domain );
+
+		foreach ( $locations as $location ) {
+			if ( ! isset( $this->cached_mo_files[ $location ] ) ) {
+				$this->set_cached_mo_files( $location );
+			}
+
+			foreach ( $this->cached_mo_files[ $location ] as $mofile ) {
+				if ( str_starts_with( str_replace( "$location/", '', $mofile ), "$domain-" ) ) {
```

Comment:
I added a `var_dump( "$domain: $mofile" );` before this line and it seems like this runs for every string. Do we need another caching layer?

<img width="1385" alt="Bildschirm­foto 2022-11-27 um 15 02 18" src="https://user-images.githubusercontent.com/617637/204139390-45a3c682-488c-49f4-81d4-d56646d7473a.png">

No other plugins active, just plain WP with 2023 theme active.

Reply:
Good point. This information can easily be cached when calling `get_path_from_lang_dir()`